servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/SecurityHandshakeObserverTest.java

```diff
@@ -204,4 +228,61 @@ private static void verifyObservers(InOrder order, TransportObserver transportOb
         }
         verifyNoMoreInteractions(transportObserver, securityHandshakeObserver);
     }
+
+    @Test
+    void optionalSslWithPlaintextDoesNotTriggerSecurityHandshake() throws Exception {
+        final CountDownLatch done = new CountDownLatch(1);
+        final AtomicBoolean securityInvoked = new AtomicBoolean(false);
+
+        try (ServerContext serverContext = BuilderUtils.newServerBuilder(SERVER_CTX)
+                .sslConfig(new ServerSslConfigBuilder(
+                        DefaultTestCerts::loadServerPem, DefaultTestCerts::loadServerKey).build())
+                .sslListenMode(SslListenMode.SSL_OPTIONAL)
+                .transportObserver((localAddress, remoteAddress) -> new ConnectionObserver() {
```

Comment:
Consider using already initialized mock `serverTransportObserver` and then verify that `serverConnectionObserver.onSecurityHandshake()` was not invoked and there were no interactions with `serverSecurityHandshakeObserver`